cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -175,6 +178,7 @@ public RepositoryMeta getRepositoryMeta(NamespaceId namespace) throws Repository
    */
   public void validateRepository(NamespaceId namespace, RepositoryConfig repoConfig)
     throws RemoteRepositoryValidationException {
+    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.GET);
```

Comment:
It might make sense here to also enforce UPDATE_REPOSITORY_METADATA as we don't expect users who aren't editing the repository to validate the config either. This will help provide a layer of protection against SSRI vulnerabilities.

Reply:
agreed, will update shortly